namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -18,6 +18,8 @@ private[consul] case class SvcKey(name: String, tag: Option[String]) {
 
 private[consul] object SvcAddr {
 
+  private[this] object ServiceRelease extends Exception with NoStackTrace
```

Comment:
what do you think about making this a `private[this] val ServiceRelease = new Exception("fjdklasf") with NoStackTrace` since SvcAddr is already a singleton anyway

Reply:
I made it an object because I was concerned that if it was a singleton `val` the equality might still get munged, but the type test would still pass?

Reply:
I think what you have right now is an equality check, not a type check.  (since ServceRelease is an object and not a type)